src/Geometry.cpp

```diff
@@ -1765,8 +1788,8 @@ void Geometry::segmentize2D(Track* track, double z_coord) {
   int num_segments;
 
   /* Use a LocalCoords for the start and end of each segment */
-  LocalCoords start(x0, y0, z0);
-  LocalCoords end(x0, y0, z0);
+  LocalCoords start(x0, y0, z0, true);
```

Comment:
I don't quite understand the definition of the class LocalCoords. Need a discussion with you.